src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -131,11 +131,49 @@ protected override void CheckInterfaces(DiagnosticBag diagnostics)
                 var conversions = new TypeConversions(corLibrary);
                 var location = singleDeclaration.NameLocation;
 
-                foreach (var @interface in interfaces)
+                foreach (var baseInterface in interfaces)
                 {
-                    @interface.CheckAllConstraints(conversions, location, diagnostics);
+                    baseInterface.CheckAllConstraints(conversions, location, diagnostics);
+                }
+
+                // we check direct base interfaces that have the InternalImplementationOnly attribute,
+                // giving an error when such an attributed interface appears in the interface list.
+                foreach (var baseInterface in this.InterfacesNoUseSiteDiagnostics(null))
```

Comment:
Correct, and the tests ensure that it is allowed to indirectly implement or extend such an interface.